common/scala/src/main/scala/whisk/core/containerpool/HttpUtils.scala

```diff
@@ -105,15 +112,33 @@ protected[core] class HttpUtils(hostname: String, timeout: FiniteDuration, maxRe
 
       response.close()
       containerResponse
+    } recoverWith {
+      // The route to target socket as well as the target socket itself may need some time to be available -
+      // particularly on a loaded system.
+      // The following exceptions occur on such transient conditions. In addition, no data has been transmitted
+      // yet if these exceptions occur. For this reason, it is safe and reasonable to retry.
+      //
+      // HttpHostConnectException: no target socket is listening (yet).
+      case t: HttpHostConnectException => Failure(RetryableConnectionError(t))
+      //
+      // NoRouteToHostException: route to target host is not known (yet).
+      case t: NoRouteToHostException => Failure(RetryableConnectionError(t))
     } match {
-      case Success(r) => r
-      case Failure(t: HttpHostConnectException) if retry =>
-        if (timeoutMsec > 0) {
-          Thread sleep 100
-          val newTimeout = timeoutMsec - 100
-          execute(request, newTimeout, retry)
+      case Success(response) => response
+      case Failure(t: RetryableConnectionError) =>
```

Comment:
Could add the `if retry` as a guard here, the `else` branch and the more general `Left` case are equivalent.